config/options.go

```diff
@@ -57,16 +59,29 @@ type Options struct {
 	// This should be used only for testing.
 	InsecureServer bool `mapstructure:"insecure_server" yaml:"insecure_server,omitempty"`
 
-	// Cert and Key is the x509 certificate used to hydrate TLSCertificate
+	// AutoCert enables fully automated certificate management including issuance
+	// and renewal from LetsEncrypt. Must be used in conjunction with CertFolder.
+	AutoCert bool `mapstructure:"autocert" yaml:"autocert,omitempty"`
+
+	// AutoCertHandler is the HTTP challenge handler used in a http-01 acme
+	// https://letsencrypt.org/docs/challenge-types/#http-01-challenge
+	AutoCertHandler func(h http.Handler) http.Handler `hash:"ignore"`
+
+	// CertFolder specifies the location to store (if using autocert), and load
+	// your TLS certificates.
+	CertFolder string `mapstructure:"certificate_folder" yaml:"certificate_folder,omitempty"`
```

Comment:
We should make the parameter name clear.  Eg, `autocert_storage_dir` or something along those lines.

Reply:
changed to `autocert_folder`